src/fidesops/util/saas_util.py

```diff
@@ -48,3 +49,32 @@ def merge_datasets(dataset: Dataset, config_dataset: Dataset) -> Dataset:
         collections=collections,
         connection_key=dataset.connection_key,
     )
+
+
+def paths_to_dict(value_map: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Converts a dictionary of paths/values into a nested dictionary
+
+    example:
+
+    {"object.first_property": "one", "object.second_property": "two"}
+
+    becomes
+
+    {
+        "object": {
+            "first_property": "one",
+            "second_property" "two"
+        }
+    }
+    """
```

Comment:
Thanks for this! Unflatten was the keyword I was missing when I was researching this. I hope you don't mind, I borrowed the error handling from your implementation and I augmented/cleaned up the test cases using yours as a reference. With that said, should this be moved back to it's own util class since this has the potential of being used outside of the SaaS work?

Reply:
either way - we can move it later if someone else wants to use it too